benchmarks/segmentation_layers.py

```diff
@@ -1,175 +1,319 @@
+
+    #
+    # cross_entropy
+    #
+    def cross_entropy_tensor_iter(self):
+        inputs, targets = self.get_input(return_targets=True)
+
+        def _cross_entropy_tensor_iter():
+            for a, b in zip(inputs, targets):
+                torch.nn.functional.cross_entropy(a.unsqueeze(0), b.unsqueeze(0)).squeeze(0)
+        return _cross_entropy_tensor_iter
+
+    def cross_entropy_tensor_pad(self):
+        inputs, targets = self.get_input(return_targets=True, pad=True)
+        i_stack = torch.stack(inputs)
+        t_stack = torch.stack(targets)
+
+        def _cross_entropy_tensor_pad():
+            torch.nn.functional.cross_entropy(i_stack, t_stack)
+
+        return _cross_entropy_tensor_pad
+
+    def cross_entropy_nt(self):
+        inputs, targets = self.get_input(return_targets=True, pad=True)
+        i_nt = nestedtensor.nested_tensor(inputs)
+        t_nt = nestedtensor.nested_tensor(targets)
+        def _cross_entropy_nt():
+            torch.nn.functional.cross_entropy(i_nt, t_nt)
+
+        return _cross_entropy_nt
+
+
+    #
+    # dropout
+    #
+    def dropout_tensor_iter(self):
+        inputs = self.get_input()
+
+        def _dropout_tensor_iter():
+            for t in inputs:
+                torch.nn.functional.dropout(t.unsqueeze(0)).squeeze(0)
+        return _dropout_tensor_iter
+
+    def dropout_tensor_pad(self):
+        inputs = self.get_input(pad=True)
+        stack = torch.stack(inputs)
+
+        def _dropout_tensor_pad():
+            torch.nn.functional.dropout(stack)
+
+        return _dropout_tensor_pad
+
+    def dropout_nt(self):
+        inputs = self.get_input()
+        nt = nestedtensor.nested_tensor(inputs)
+        def _dropout_nt():
+            torch.nn.functional.dropout(nt)
+
+        return _dropout_nt
+        inputs = get_input_big_diff()
+        nt = nestedtensor.nested_tensor(inputs)
+
+        def _dropout_nt_big_diff():
+            dropout(nt)
+
+        return _dropout_nt_big_diff
+
+    #
+    # interpolate
+    #
+    def interpolate_tensor_iter(self):
+        inputs = self.get_input()
+
+        def _interpolate_tensor_iter():
+            for t in inputs:
+                torch.nn.functional.interpolate(t,  t.unsqueeze(0).shape[-2])
+        return _interpolate_tensor_iter
+
+    def interpolate_tensor_pad(self):
+        inputs = self.get_input(pad=True)
+        stack = torch.stack(inputs)
+
+        def _interpolate_tensor_pad():
+            torch.nn.functional.interpolate(stack, inputs[0].unsqueeze(0).shape[-2])
+
+        return _interpolate_tensor_pad
+
+    def interpolate_nt(self):
+        inputs = self.get_input()
+        nt = nestedtensor.nested_tensor(inputs)
+        def _interpolate_nt():
+            torch.nn.functional.interpolate(nt)
+
+        return _interpolate_nt
+
+def main(args):
+    assert len(args) == 9
+    name = args[0]
+    benchmark_obj = SegLayersBenchMark(args[1:])
+    benchmark = getattr(benchmark_obj, name)
```

Comment:
You can explicitly use a dictionary instead of getting the method of the class and pass a configuration object as an argument.

It's pretty much the same thing though, just a different way to organize.